LethalAvatars/AvatarDriver.cs

```diff
@@ -244,10 +245,23 @@ internal void AnimationDone(Animator anim, Transform camera, bool isLocal)
         avatarNearClips.ForEach(x => x.CreateShadows());
     }
 
+    internal void SetupAvatar()
+    {
+        Transform? head = GetBoneFromHumanoid(HumanBodyBones.Head);
```

Comment:
Given the review about this method being invoked before `InitializeAnimatorControllers(Avatar)`, you will need to change this method to check the Animator itself, otherwise an exception will be thrown that will break the whole avatar.

Here is how I implemented this

```cs
internal void SetupAvatar(Animator anim)
{
    if(anim == null || anim.avatar == null) return;
    Transform? head = anim.GetBoneTransform(HumanBodyBones.Head);
    if (head == null || Avatar == null) return;
    // Assume the root gameobject points forward
    Quaternion a = head.rotation;
    Quaternion b = Avatar.transform.rotation;
    Quaternion delta = Quaternion.Inverse(b) * a;
    headRef = delta;
}
```